src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -425,22 +452,30 @@ private async Task FlushSettingsUpdates()
             }
         }
 
-        private async Task UpdateCatagory(Guid categoryId, ExceptionCategorySettings categorySettings, SettingsUpdates updates)
+        private async Task UpdateCategory(Guid categoryId, ExceptionCategorySettings categorySettings, SettingsUpdates updates)
         {
             // Update the category
-            if (updates.NewCategoryState.HasValue && (
-                updates.NewCategoryState.Value != ExceptionBreakpointStates.None || // send down a rule if the category isn't in the default state
-                categorySettings.CurrentRules.Count != 0)) // Or if we have other rules for the category that we need to blow away
+            lock (categorySettings.CurrentRules)
```

Comment:
Move this lock to be just around 
`categorySettings.CurrentRules.Clear();` and `categorySettings.CurrentRules.Add("*", breakpointId);` so you can use await on `_commandFactory`.